channels/offline-channel-js/src/Sender.ts

```diff
@@ -0,0 +1,717 @@
+        let _endpointUrl: string;
+        let _enableSendPromise: boolean;
+        let _alwaysUseCustomSend: boolean;
+        let _disableXhr: boolean;
+        let _fallbackSend:  SendPOSTFunction;
+        let _isInitialized: boolean;
+        let _diagLog: IDiagnosticLogger;
+        let _core: IAppInsightsCore;
+        let _httpInterface: IXHROverride;
+        let _onlineChannelId: string;
+        let _isOneDs: boolean;
+       
+
+        dynamicProto(Sender, this, (_self, _base) => {
+
+            let _sendCredentials = true; // for 1ds
+            _initDefaults();
+
+            _self.pause = () => {
+                _clearScheduledTimer();
+                _paused = true;
+            };
+        
+            _self.resume = () => {
+                if (_paused) {
+                    _paused = false;
+                    _retryAt = null;
+                }
+            };
+
+            _self.getXhrInst = (sync?: boolean): IXHROverride => {
+                // unload events will be saved. so not return unload interface
+                return _httpInterface;
+            }
+        
+            _self.initialize = (config: IConfiguration & IConfig, core: IAppInsightsCore, cxt: IProcessTelemetryContext, diagLog: IDiagnosticLogger,  channelId?: string, unloadHookContainer?: IUnloadHookContainer): void => {
+                
+                _diagLog = diagLog || core.logger;
+                if (_isInitialized) {
+                    _throwInternal(_diagLog, eLoggingSeverity.CRITICAL, _eInternalMessageId.SenderNotInitialized, "Sender is already initialized");
+                }
+                _core = core;
+                _consecutiveErrors = 0;
+                _retryAt = null;
+                _stamp_specific_redirects = 0;
+
+                // This function will be re-called whenever any referenced configuration is changed
+                let hook = onConfigChange(config, (details) => {
+                    let config = details.cfg;
+                    if (config.storagePrefix){
+                        utlSetStoragePrefix(config.storagePrefix);
+                    }
+                    let ctx = createProcessTelemetryContext(null, config, core);
+                   
+                    let offlineCfg = ctx.getExtCfg(DefaultOfflineIdentifier) as ILocalStorageConfiguration;
+                    _onlineChannelId = channelId || BreezeChannelIdentifier;
+                    let senderConfig = ctx.getExtCfg(_onlineChannelId, {}) as any;
+                    let offlineSenderCfg = offlineCfg.senderCfg || {} as IOfflineSenderConfig;
+                  
+                    if (_onlineChannelId == PostChannelId) {
+                        _isOneDs = true;
+                    }
+                    
+                    _endpointUrl = endpointUrl || senderConfig.endpointUrl || config.endpointUrl || DEFAULT_BREEZE_ENDPOINT + DEFAULT_BREEZE_PATH;
```

Comment:
As above lets remove this as a fallback if the payload doesn't provide an endpoint we just don't send / return an error.